### PR TITLE
Make maxTaskStringSize configurable

### LINKED ISSUE #716: Individual Task Query while using tasktracker executor cannot exceed 12kb
When the user runs a `SELECT` query and has the executor type set to `task-tracker`, the master node breaks up the query into query fragments and sends query fragments to worker nodes.  On each worker, there's a daemon process. The daemon process assigns individual query fragments to itself to execute. There is a limit on the length of these query fragments, they cannot exceed 12kb. (edited)
If the individual task query exceeds 12kb, then we get this error:

```
WARNING: could not receive query results from <worker-ip>:5432
DETAIL: Client error: task call string exceeds maximum assignable length
```

All the individual task queries are stored in shared memory and hence we put hard limit on what length the individual task query can be.

### PR DESCRIPTION
**Pull request by mtuncer**
taskCallStringSize determines the size of worker query string.
It was originally hard coded to a specific value. This has caused
issues at some users. Since it determines initial shared memory
allocation, we did not want to set it to an arbitrary higher number.
Instead made it configurable.

This commit introduces a new GUC variable max_task_string_size

Changes in this variable requires restart to be in effect.

Fixes #716

### CODE REVIEW

File: src/backend/distributed/worker/task_tracker.c [hunk 1]
```diff
@@ -662,9 +664,10 @@ SchedulableTaskList(HTAB *WorkerTasksHash)
 
 	for (queueIndex = 0; queueIndex < tasksToScheduleCount; queueIndex++)
 	{
-		WorkerTask *schedulableTask = (WorkerTask *) palloc0(sizeof(WorkerTask));
-		schedulableTask->jobId = schedulableTaskQueue[queueIndex].jobId;
-		schedulableTask->taskId = schedulableTaskQueue[queueIndex].taskId;
+		WorkerTask *schedulableTask = (WorkerTask *) palloc0(WORKER_TASK_SIZE);
+		WorkerTask *queuedTask = WORKER_TASK_AT(schedulableTaskQueue, queueIndex);
```
Comment (mtuncer) [inline src/backend/distributed/worker/task_tracker.c]: Not use if this is the best way to access array items. Open to suggestions

File: src/include/distributed/task_tracker.h [hunk 2]
```diff
@@ -72,21 +75,28 @@ typedef enum
...
 	char userName[NAMEDATALEN]; /* user to use for local backend connection */
 	int32 connectionId;     /* connection id to local backend */
 	uint32 failureCount;    /* number of task failures */
+	char taskCallString[FLEXIBLE_ARRAY_MEMBER]; /* query or function call string */
 } WorkerTask;
 
+#define WORKER_TASK_SIZE (offsetof(WorkerTask, taskCallString) + \
+						  MaxWorkerTaskCallStringSize)
+#define WORKER_TASK_AT(workerTasks, index) \
+	((WorkerTask *) (((char *) (workerTasks)) + (index) * WORKER_TASK_SIZE))
```
Comment (mtuncer) [inline src/include/distributed/task_tracker.h]: Working around pointer arithmetic warning.

**Comment by samay-sharma** [comment]
hey @mtuncer : The values look good to me. Could we rename the guc to something shorter?

Suggestions: max_task_string_size. We generally don't have master / worker in guc names. We document that info in docs and in the description. Also, I don't know if "TaskCall" is something which users would understand.

Also, could we add a hint to increase this config to the place where the user would previously get "task call string exceeds maximum assignable length". Also, maybe rephrase current message to "task string exceeds .... length [size]"

**Comment by mtuncer** [comment]
@anarazel can you take a look

File: src/include/distributed/task_tracker.h [hunk 3]
```diff
@@ -72,21 +72,28 @@ typedef enum
...
+ * Since taskCallString is dynamically sized use WORKER_TASK_SIZE instead of
+ * sizeof(WorkerTask). Use WORKER_TASK_AT to reference an item in WorkerTask array.
  */
 typedef struct WorkerTask
 {
 	uint64 jobId;      /* job id (upper 32-bits reserved); part of hash table key */
 	uint32 taskId;     /* task id; part of hash table key */
 	uint32 assignedAt; /* task assignment time in epoch seconds */
 
-	char taskCallString[TASK_CALL_STRING_SIZE]; /* query or function call string */
```
Comment (pykello) [inline src/include/distributed/task_tracker.h]: Also remove ```#define TASK_CALL_STRING_SIZE``` from task_tracker.h?
